frontend_tests/node_tests/hotkey.js

```diff
@@ -292,7 +307,7 @@ run_test("streams", ({override}) => {
 run_test("basic mappings", () => {
     assert_mapping("?", browser_history, "go_to_location");
     assert_mapping("/", search, "initiate_search");
-    assert_mapping("w", activity, "initiate_search");
+    assert_mapping_rewire("w", activity, "initiate_search");
```

Comment:
This is the only use of `assert_mapping_rewire` and `stubbing_rewire`. All the other `assert_mapping` calls take mocked modules as arguments. But `activity` is not mocked—is that intended?

Reply:
The reason that we treat `activity` differently basically comes down to the fact that `hotkey` tells `activity` every time a key is pressed, so that we can send the appropriate bool to the `/json/users/me/presence` endpoint.

~~~ js
943 export function process_keydown(e) {
944     activity.set_new_user_input(true);
945     const hotkey = get_keydown_hotkey(e);
946     if (!hotkey) {
947         return false;
948     }
949     return process_hotkey(e, hotkey);
950 }
~~~

This will be an interesting case study, and this is a good example of how your strategy in this PR adds some additional clarity on where we should be hunting down code smells. (This code seems smelly, but not stinky.  I am not sure I'd make heroic efforts to change the actual code here, but there might be a better way to handle it in the test code.  Not really sure until I dive deeper.)

Reply:
I **think** what this test is trying to tell us is that we should split out a little data module to track new user input, rather than putting that responsibility into `activity.js`, which is one of our earliest kitchen-sink legacy modules with too many responsibilities.  But, again, I would need to dive deeper to be sure.